### PR TITLE
Check for Organization when Determining if Collective is a Fiscal Host

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/5497

fix-independent-collective

### CODE REVIEW

File: components/edit-collective/actions/Archive.js [hunk 1]
```diff
@@ -49,7 +50,7 @@ const ArchiveCollective = ({ collective }) => {
   };
   const [archiveCollective] = useMutation(archiveCollectiveMutation, adminPanelMutationParams);
   const [unarchiveCollective] = useMutation(unarchiveCollectiveMutation, adminPanelMutationParams);
-  const isSelfHosted = collective.host?.id === collective.id;
+  const isSelfHosted = collective.host?.id === collective.id && collective.type !== CollectiveType.ORGANIZATION;
```
Comment (znarf) [inline components/edit-collective/actions/Archive.js:53]:
```suggestion
  const isIndependentCollective = collective.host?.id === collective.id && collective.type == CollectiveType.COLLECTIVE;
```

**Review by znarf — COMMENTED** [review]
In the end, when I look at it again, I barely understand why we have this error message.

When archiving an Independent Collective, resetting Fiscal Host setting should be automated, like any other Collective.

File: components/edit-collective/actions/Archive.js (hunk 1 above)
Comment (znarf) [reply components/edit-collective/actions/Archive.js:53]: Can we please move forward there?

Comment (SudharakaP) [reply components/edit-collective/actions/Archive.js]: Sorry, somehow seems I've missed this one. I've added the `isIndependentCollective` condition. But I don't see where we should use the `isHost` condition as we already have `collective.isHost`;

https://github.com/opencollective/opencollective-frontend/blob/29989675f9fd16748adad5bff28a70535a5507d7/components/edit-collective/actions/Archive.js#L153

So if the collective is not an independent collective and `collective.isHost` is true, then it should be a host (`isHost=true`).

**Comment by SudharakaP** [comment]
> In the end, when I look at it again, I barely understand why we have this error message.
> 
> When archiving an Independent Collective, resetting Fiscal Host setting should be automated, like any other Collective.

Makes sense. I'll reiterate this on a new issue; https://github.com/opencollective/opencollective/issues/5543

File: components/edit-collective/actions/Archive.js [hunk 2]
```diff
@@ -150,7 +152,7 @@ const ArchiveCollective = ({ collective }) => {
       )}
       {!isArchived && collective.isHost && (
         <P color="rgb(224, 183, 0)" my={1}>
-          {isSelfHosted ? (
+          {isIndependentCollective ? (
```
Comment (znarf) [inline components/edit-collective/actions/Archive.js:155]:
```suggestion
          {collective.type === CollectiveType.COLLECTIVE ? (
```

That would be simpler.

Comment (znarf) [reply components/edit-collective/actions/Archive.js:155]: You can remove the `isIndependentCollective` variable.

File: components/edit-collective/actions/Archive.js (hunk 1 above)
Comment (znarf) [reply components/edit-collective/actions/Archive.js]: I looked at it again and proposed another simplification.

File: components/edit-collective/actions/Archive.js (hunk 2 above)
Comment (SudharakaP) [reply components/edit-collective/actions/Archive.js]: This is done. 👍🏽

**Review by znarf — APPROVED** [review]

**Pull request merged**